cypher-manual/extensions.md

```diff
@@ -0,0 +1,50 @@
+---
+id: extension-clauses
+title: Memgraph Cypher extensions 
+sidebar_label: Memgraph Cypher extensions
+---
+
+### Foreach
+
+***foreach*** iterates over a list of elements. Each element is stored inside a variable which can
+optionally be used inside the update clauses. All update clauses are executed per iteration of the list.
+
+```cypher
+  FOREACH ( <variable name> IN <expression> | <update clauses> )
+```
+|         Option          |                                                         Description                                                       |
+| :---------------------: | :-----------------------------------------------------------------------------------------------------------------------: |
+|       variable name     |                                         The variable name that stores each element                                        |
+|        expression       |                                           Any expression that results to a list                                           |
+|      update clauses     |                  One or more cypher update clauses: SET, REMOVE, CREATE, MERGE, DELETE including FOREACH extension        |
+
+It must be noted that if the result &lt;expression&gt; is null, then **foreach** will not fail but rather skip the execution of &lt;update clauses&gt; alltogether.
+
+Examples: 
+
+```cypher
+  FOREACH ( i IN [1, 2, 3] | CREATE (n {id : i}) )
+```
+Creates 3 nodes, each with the id property set to 1, 2 and 3 respectively.
+
+
+```cypher
+  CREATE (n { prop : [[1, 2], [3, 4]]);
+
+  MATCH (n) FOREACH ( i IN n.prop | FOREACH ( j IN i | CREATE (u { prop : j }) ) );
```

Comment:
```suggestion
  MATCH (n) FOREACH ( inner_list IN n.prop | FOREACH ( i IN inner_list | CREATE (u { prop : i }) ) );
```

Maybe gives a bit more context. I am not happy with `inner_list`, but I think it is important to give it a more explanatory name than `i`.